### PR TITLE
[MLIR]Reuse one tuning parameters

### PR DESCRIPTION
**Pull request by yiqian1**
This PR will pass a new tuning parameter to MLIR to help tuning MLIR kernels.

This patch affects solvers ConvMlirIgemmFwdXdlops, ConvMlirIgemmBwdXdlops and ConvMlirIgemmWrWXdlops. The tuning time for these solvers will double. The average improvement for Xdlops will be about 10%. There is no changes on the non-Xdlops side.

The MLIR library that uses this newly passed tuning parameter will be in the ROCm5.5 release. The MLIR library older than ROCm5.5 RC is not affected by this PR. That means there is no need to re-tune those solvers in MLIR library that is older than ROCm5.5 RC.

### CODE REVIEW

**Comment by atamazov** [comment]
@yiqian1 
> This PR is just for review. Don't merge.

Please convert it into draft then; that should block merging.

**Comment by jerryyin** [comment]
This diff looks good to me. I think you are doing the experiment correctly from MIOpen end.

**Comment by atamazov** [comment]
The following questions must be answered prior merging this:
- How this affects tuning for MLIR solvers? Expectations:
  - Tuning time doubles.
  - Tuning efficiency improves.
- What is the real average improvement of tuning efficiency?
  - Answering this question required extensive testing
- Am I correct that existing perf-db contents is compatible with updated MLIR solvers? In other words, the re-tuning of MLIR solvers is desirable (to obtain slightly better performance from MLIR kernels) but not required, right?
- Which version of MLIR library is capable to correctly use the newly passed tuning parameter?

**Comment by yiqian1** [comment]
> The following questions must be answered prior merging this:
> 
> * How this affects tuning for MLIR solvers? Expectations:
>   
>   * Tuning time doubles.
>   * Tuning efficiency improves.

Tuning time for Xdlops will double. The average improvement for Xdlops will be about 10%. No changes on non-Xdlops.

> * What is the real average improvement of tuning efficiency?
>   
>   * Answering this question required extensive testing
> * Am I correct that existing perf-db contents is compatible with updated MLIR solvers? In other words, the re-tuning of MLIR solvers is desirable (to obtain slightly better performance from MLIR kernels) but not required, right?

Yes, you're correct.

> * Which version of MLIR library is capable to correctly use the newly passed tuning parameter?

I'll merge the MLIR changes after this PR is in. The MLIR library that uses this new parameter will be in the Rocm5.5 release.

**Comment by atamazov** [comment]
@yiqian1 ...and the current MLIR library correctly ignores the new parameter, right?

❗ I recommend updating the description so any visitor is able to quickly understand the expected side effects of this PR.

**Comment by atamazov** [comment]
@yiqian1 
> yiqian1 force-pushed the tune_unroll_indexdiff branch from e59bb09 to 5e63346 

⚠️ Please read: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1826#issuecomment-1284379890

**Review by atamazov — APPROVED** [review]
No objections.

**Review by jerryyin — APPROVED** [review]

**Comment by yiqian1** [comment]
@atamazov Could you please help merge this PR?

**Comment by atamazov** [comment]
@yiqian1 The PR description is still missing the explanation of motivation. In other words, the expected side effects are not described.

Please add the following labels:
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/performance
- I highly recommend using https://github.com/ROCmSoftwarePlatform/MIOpen/labels/IMPACTS_PERFORMANCE_DB instead of https://github.com/ROCmSoftwarePlatform/MIOpen/labels/tuning - the latter should be more about common tuning machinery than about solver changes that affect perf-db.
- Appropriate `urgency` label, e.g. https://github.com/ROCmSoftwarePlatform/MIOpen/labels/urgency_normal

Please also clearly list the names of the affected Solvers there and clearly mention that _re-tuning of these Solvers is meaningless with MLIR library older than ROCm 5.5 RC_. This information is necessary for the Tuna team.

/cc @cderb @JehandadKhan @junliume

**Comment by atamazov** [comment]
> @atamazov Could you please help merge this PR?

Unfortunately, I do not have enough rights as of now. Please ask @junliume.

**Comment by yiqian1** [comment]
@junliume Can you or someone else help me to merge this PR please?

**Comment by jerryyin** [comment]
FYI: This PR will be blocking the ROCm 5.5 release of rocMLIR but it is not dependent on commit hash of MLIR. (and perfdb will be backward compatible)

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@yiqian1 Now the description is good, thank you!

**Pull request merged**
